content/4_Gold/Modular.mdx

```diff
@@ -263,6 +266,65 @@ public class main
 
 </LanguageSection>
 
+### Solution - Binomial Coefficients
+
+In order to calculate $a \choose b$, we can precalculate the factorials to speed up the calculation of $a!$, $b!$, and $(a-b)!$. $a \choose b$ is equivalent to
+
+$$
+\frac{a!}{b!(a-b)!} = a! \cdot b!^{-1} \cdot (a-b)!^{-1}.
+$$
+
+Thus, calculating the inverses of $b!$ and $(a-b)!$ is sufficient in finding $a \choose b$.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+ 
+using ll = long long;
+const int MOD = (int)1e9+7;
+ 
+ll inverse(ll num){
+    ll a = num; ll b = MOD-2;
+    //equivalent to num^(MOD-2)
+    ll res = 1;
+    while (b > 0) {
+        if (b % 2 == 1)
+            res = res * a % MOD;
+        a = a * a % MOD;
+        b /= 2;
+    }
+    return res;
+}
+ 
+int main() {
+    cin.tie(0)->sync_with_stdio(0);
+    ll factorial[1000001];
+    factorial[0] = factorial[1] =1;
+    for (int i = 2; i<=1000000; i++) {
+        factorial[i]=(factorial[i-1]*i)%MOD;
+    }
+ 
+    ll n, a, b;
+    cin >> n;
+    for(int i = 0; i < n; i++) {
+        cin >> a >> b;
+        ll ans = (factorial[a] * inverse(factorial[b])) % MOD;
```

Comment:
worth noting that if you compute `inverse(factorial[n])` then you can get `inverse(factorial[1..n-1])` in O(n) time.